lib/shopify_transporter/exporters/magento/magento_exporter.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require_relative './customer_exporter.rb'
+require_relative './order_exporter.rb'
+
+module ShopifyTransporter
+  module Exporters
+    module Magento
+      class MissingExporterError < ExportError; end
+
+      class MagentoExporter
+        def self.for(type, store_id, client)
+          case type
+          when 'customer'
+            CustomerExporter
+          when 'order'
+            OrderExporter
+          else
+            raise MissingExporterError
```

Comment:
We don't seem to be rescuing this anywhere.

Reply:
I don't believe that _should_ be rescued anywhere.

Thor knows that it expects either `customer` or `order` as the object type. If it receives something else, that is handled by Thor:

```
$ exe/shopify_transporter export --object=bloop --config=config.yml
Expected '--object' to be one of customer, order; got bloop
```

It is truly an exceptional circumstance if an unexpected parameter gets through Thor's check and makes it to this function. So, I think it is reasonable for an exception to get bubbled up to the top level.

One change I would maybe accept is to just not even raise an error in this case, since it literally should never happen

Reply:
I'm okay with raising and not rescuing here - we do want this to blow up if an invalid option got this far, and this gives us a specific error when debugging/solving it.